design/api/build-spi/swagger.yaml

```diff
@@ -0,0 +1,431 @@
+      parameters:
+      - $ref: "#/parameters/BuildRequest"
+      responses:
+        200:
+          description: "OK"
+          schema:
+            type: object
+            properties:
+              BuildName:
+                type: string
+        401:
+          $ref: "#/responses/401"
+        500:
+          $ref: "#/responses/500"
+  /namespace/{namespace}/builds/{name}:
+    parameters:
+    - $ref: "#/parameters/Namespace"
+    - $ref: "#/parameters/Name"
+    get:
+      summary: "Get build status."
+      description: "Get build status."
+      responses:
+        200:
+          description: "OK"
+          schema:
+            $ref: "#/definitions/BuildStatus"
+        401:
+          $ref: "#/responses/401"
+        500:
+          $ref: "#/responses/500"
+    delete:
+      summary: "Delete a Build."
+      description: "Delete a Build."
+      responses:
+        204:
+          $ref: "#/responses/204"
+        401:
+          $ref: "#/responses/401"
+        500:
+          $ref: "#/responses/500"
+  /namespace/{namespace}/builds/{name}/cancel:
+    parameters:
+    - $ref: "#/parameters/Namespace"
+    - $ref: "#/parameters/Name"
+    post: 
+      summary: "Cancel a Build."
+      description: "Cancel a Build."
+      responses:
+        204:
+          $ref: "#/responses/204"
+        401:
+          $ref: "#/responses/401"
+        500:
+          $ref: "#/responses/500"
+  /namespace/{namespace}/builds/{name}/artifacts:
+    parameters:
+    - $ref: "#/parameters/Namespace"
+    - $ref: "#/parameters/Name"
+    get:
+      summary: "List artifacts."
+      description: "List artifacts."
+      responses:
+        200:
+          description: "OK"
+          schema:
+            $ref: "#/definitions/ArtifactList"
+        401:
+          $ref: "#/responses/401"
+        405:
+          $ref: "#/responses/405"
+        500:
+          $ref: "#/responses/500"
+  /namespace/{namespace}/builds/{name}/artifacts/{artifactid}:
+    parameters:
+    - $ref: "#/parameters/Namespace"
+    - $ref: "#/parameters/Name"
+    - $ref: "#/parameters/ArtifactId"
+    get:
+      summary: "Get artifact."
+      description: "Get artifact."
+      produces:
+      - "*/*"
+      responses:
+        200:
+          description: "OK"
+        401:
+          $ref: "#/responses/401"
+        404:
+          $ref: "#/responses/404"
+        405:
+          $ref: "#/responses/405"
+        500:
+          $ref: "#/responses/500"
+  /namespace/{namespace}/builds/{name}/log:
+    parameters:
+    - $ref: "#/parameters/Namespace"
+    - $ref: "#/parameters/Name"
+    - $ref: "#/parameters/Follow"
+    get:
+      summary: "Get logs."
+      description: "Get logs."
+      produces:
+      - "text/plain; charset=utf-8"
+      responses:
+        200:
+          $ref: "#/responses/200"
+        401:
+          $ref: "#/responses/401"
+        500:
+          $ref: "#/responses/500"
+  /namespace/{namespace}/builds/{name}/stage/{stageid}/log:
```

Comment:
@rupalibehera well, in theory you could have `/namespace/{namespace}/builds/{name}/stage//log` (notice `//`) instead of `/namespace/{namespace}/builds/{name}/log` but that's just not a **nice** REST API. I feel that separating it into 2 paths makes it easier for people to understand